test/e2e/reconciler/cache/replication_test.go

```diff
@@ -60,109 +61,264 @@ type testScenario struct {
+	require.NoError(t, err)
+	t.Logf("Verify that APIExport %s/%s is propaged to the cached object after an update", cluster, export.Name)
+	verifyResourceExistence(t, scenario, cluster)
+}
+
+func verifyResourceExistence(t *testing.T, scenario baseScenario, cluster logicalcluster.Name) {
+	t.Logf("Get %s %s/%s from the root shard and the cache server for comparison", scenario.resourceKind, cluster, scenario.resourceName)
 	framework.Eventually(t, func() (bool, string) {
-		wildAPIExport, err = kcpShardClusterClient.Cluster(cluster).ApisV1alpha1().APIExports().Get(ctx, "wild.wild.west", metav1.GetOptions{})
+		originalResource, err := scenario.getSourceResource(cluster)
 		if err != nil {
 			return false, err.Error()
 		}
-		cachedWildAPIExport, err = cacheKcpClusterClient.Cluster(cluster).ApisV1alpha1().APIExports().Get(cacheclient.WithShardInContext(ctx, shard.New("root")), wildAPIExport.Name, metav1.GetOptions{})
+		cachedResource, err := scenario.getCachedResource(cluster)
 		if err != nil {
 			if !errors.IsNotFound(err) {
-				return false, err.Error()
+				return true, err.Error()
 			}
 			return false, err.Error()
 		}
-		t.Logf("Verify if both the orignal APIExport and replicated are the same except %s annotation and ResourceVersion after creation", genericapirequest.AnnotationKey)
-		if _, found := cachedWildAPIExport.Annotations[genericapirequest.AnnotationKey]; !found {
-			t.Fatalf("replicated APIExport root/%s/%s, doesn't have %s annotation", cluster, cachedWildAPIExport.Name, genericapirequest.AnnotationKey)
+		t.Logf("Verify if both the orignal and replicated resources (%s %s/%s) are the same except %s annotation and ResourceVersion after creation", scenario.resourceKind, cluster, scenario.resourceName, genericapirequest.AnnotationKey)
+		cachedResourceMeta, err := meta.Accessor(cachedResource)
+		if err != nil {
+			return false, err.Error()
+		}
+		if _, found := cachedResourceMeta.GetAnnotations()[genericapirequest.AnnotationKey]; !found {
+			t.Fatalf("replicated %s root|%s/%s, doesn't have %s annotation", scenario.resourceKind, cluster, cachedResourceMeta.GetName(), genericapirequest.AnnotationKey)
 		}
-		delete(cachedWildAPIExport.Annotations, genericapirequest.AnnotationKey)
-		if diff := cmp.Diff(cachedWildAPIExport, wildAPIExport, cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion")); len(diff) > 0 {
-			return false, fmt.Sprintf("replicated APIExport root/%s/%s is different that the original", cluster, wildAPIExport.Name)
+		delete(cachedResourceMeta.GetAnnotations(), genericapirequest.AnnotationKey)
+		if diff := cmp.Diff(cachedResource, originalResource, cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion")); len(diff) > 0 {
+			return false, fmt.Sprintf("replicated %s root|%s/%s is different that the original", scenario.resourceKind, cluster, cachedResourceMeta.GetName())
 		}
 		return true, ""
 	}, wait.ForeverTestTimeout, 400*time.Millisecond)
```

Comment:
How did we arrive at this poll period?

Reply:
just a random number

Reply:
Let's use 100ms like the rest of the tests unless we really need to change it - for folks with fast workstations, 300ms extra poll time might double or triple test runtime :)

Reply:
> for folks with fast workstations, 300ms extra poll time might double or triple test runtime :)

that's why you want to use `t.Parallel` everywhere :)

Reply:
Hey, what takes 65 minutes in GitHub Actions takes 54seconds on my machine ... ;)